doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,141 @@
+| `MoveSelectionAnchorDownOneScreen`   | `Shift` + `Page Down`      | `Down`    , `Viewport`
+| `MoveSelectionAnchorToLeftMargin`    | `Shift` + `Home`           | `Left`    , `Viewport`
+| `MoveSelectionAnchorToRightMargin`   | `Shift` + `End`            | `Right`   , `Viewport`
+| `MoveSelectionAnchorUpByWord`        | `Ctrl` + `Shift` + `↑`     | `Up`      , `Word`
+| `MoveSelectionAnchorDownByWord`      | `Ctrl` + `Shift` + `↓`     | `Down`    , `Word`
+| `MoveSelectionAnchorLeftByWord`      | `Ctrl` + `Shift` + `←`     | `Left`    , `Word`
+| `MoveSelectionAnchorRightByWord`     | `Ctrl` + `Shift` + `→`     | `Right`   , `Word`
+| `MoveSelectionAnchorToBufferStart`   | `Ctrl` + `Shift` + `Home`  | N/A
+| `MoveSelectionAnchorToBufferEnd`     | `Ctrl` + `Shift` + `End`   | N/A
+| `SelectEntireBuffer`                 | `Ctrl` + `A`               | N/A
+| `ToggleMarkMode`                     | `Ctrl` + `M`               | N/A
+
+
+## Capabilities
+
+### Accessibility
+
+Being able to create and update a selection without the use of a mouse is a very important feature to users with disabilities. The UI Automation system (particularly with the use of the signaling model), should be aware that a new selection exists. Thus, the bounding rects should be updated and the associated text regions should be reread appropriately. 
+
+### Security
+
+N/A
+
+### Reliability
+
+With regards to the Terminal Core, the newly introduced code should rely on already existing and tested code. Thus no crash-related bugs are expected.
+
+With regards to Terminal Control and the settings model, crash-related bugs are not expected. However, ensuring that the selection is updated and cleared in general use-case scenarious must be ensured.
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+The settings model makes all of these features easy to disable, if the user wishes to do so. So, no.
+
+## Future considerations
+
+### Impact from Keybinding Args
+A lot of the introduced keybindings will have a significant impact from the Keybinding Args feature. This will make most of the keybindings collapse to the following format:
+
+```JS
+{
+    "command": "MoveSelectionAnchor",
+    "keybinding": "Shift + Up",
+    "args": {
+        "direction": "Up",
+        "expansionMode": "Cell"
+    }
+}
+```
+
+The introduced `enum direction` and `enum expansionMode` args should be able to handle all of the `MoveSelectionAnchor...` commands.
+
+### Expanding Mark Mode functionality
+This spec currently limits mark mode functionality to the following:
+- only move by cell (no other expansion modes)
+- no rebinding arrow keys allowed
+In addition to the `enum direction` and `enum expansionMode` args introduced above, a `bool markMode` arg can be added. If set to `true`, the selection anchor will adjust to these commands when in mark mode. This will allow the selection anchor to be able to be adjusted by different expansion modes.

```

Comment:
I don't think I understand this paragraph - is it saying these actions will _only_ work in mark mode when `markMode:true` is set, or is it saying they'll _additionally_ work in mark mode?

Should true be the default here, or false?

Reply:
This is tough. Original mark mode works well in that _holding_ shift moves the second selection anchor, otherwise move the cursor. Here, we can't have that unless shift is on all of the keybindings (which we can't guarantee).

We could let users rebind moving the cursor in mark mode? That way they don't have to use the arrow keys (maybe use WASD?). But we'd have to do something like `MarkMode_MoveCursor*`. So, I guess the approach for this idea to work would be `bool markMode` means that this keybinding only moves the cursor when in mark mode?

Reply:
Have you considered following tmux's implementation of this feature?  It has the benefit of being well-known, and it won't step on tmux's own implementation as long as the key combination to activate it doesn't coincide with tmux's

Here's a little guide that explains how that works:
http://www.rushiagr.com/blog/2016/06/16/everything-you-need-to-know-about-tmux-copy-pasting-ubuntu/

Reply:
> Have you considered following tmux's implementation of this feature? It has the benefit of being well-known, and it won't step on tmux's own implementation as long as the key combination to activate it doesn't coincide with tmux's
> 
> Here's a little guide that explains how that works:
> http://www.rushiagr.com/blog/2016/06/16/everything-you-need-to-know-about-tmux-copy-pasting-ubuntu/

Huh. That sounds like a modification to mark mode. It'd be easy to implement too. Idk if that should be an extension or an in-box option (like `copyOnSelect`). I'll file a separate issue to track this. Thanks! 😊

Reply:
Wait, I'm dumb (loooong day). This would be a good workaround for the problem. Polling for thoughts from the team.

Reply:
@miniksa @DHowett-MSFT @cinnamon-msft

Reply:
I think the conhost style was a bit more intuitive than the tmux one. And the author of that blog post seems to agree with me based on the fact that they wrote a blog post to explain how to do it and noted that everyone struggles with copying in tmux at some point. But I'm up for whatever the community thinks is more useful.

Reply:
Ok. I'll be updating the spec soon but the short version of the design will be as follows:
- by default, mark mode operates like it does in conhost
- a new keybinding will be added to make it so that if you press it, you toggle which endpoint you're moving (similar to tmux)